sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1966,7 +1966,20 @@ case class ElementAt(left: Expression, right: Expression)
   }
 
   override def nullable: Boolean = left.dataType match {
-    case _: ArrayType => computeNullabilityFromArray(left, right)
+    case _: ArrayType =>
+      def specialNormalizeIndex: (Int, Int) => Int = {
+        (arrayLength: Int, index: Int) => {
+          if (index < 0) {
+            arrayLength + index
+          } else if (index == 0) {
```

Comment:
why not
```
if (index <= 0) {
  arrayLength + index
} ...
```

Reply:
Actually, `ElementAt` fails at runtime if `index == 0`, so the nullable doesn't really matter.

Reply:
but if the passed in index is 0, it will change to -1 and call the following code. it will throw exception, but the old behavior is return a default true. 
```
ar(intOrdinal).nullable
```

Reply:
I am just try to follow the old behavior.